### PR TITLE
Add extensive unit and component tests for scheduling activities

### PR DESCRIPTION
**Pull request by sfmskywalker**
- Introduced unit tests for `Cron`, `Delay`, `Timer`, and `StartAt` scheduling activities, covering general usage and corner cases.
- Added component tests validating `Cron`, `Delay`, `Timer`, and `StartAt` workflows within broader scenarios, focusing on workflow execution, blocking, and resumption of activities.
- Enhanced test project structures with new folder setups aligning to activity categories.
- Updated namespaces and project files to match the new structure and added validation for scheduling logic.

- - -
This change is [Reviewable](https://reviewable.io/reviews/elsa-workflows/elsa-core/7036)

### CODE REVIEW

File: test/unit/Elsa.Activities.UnitTests/Scheduling/CronTests.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        // Arrange
+        var expectedTime = new DateTimeOffset(2025, 1, 7, 0, 0, 0, TimeSpan.Zero);
+        var cronParser = CreateCronParser(expectedTime);
+        var activity = Cron.FromCronExpression("0 0 0 * * *");
+
+        // Act
+        var context = await ExecuteAsync(activity, cronParser);
+
+        // Assert
+        Assert.True(context.JournalData.ContainsKey("ExecuteAt"));
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply test/unit/Elsa.Activities.UnitTests/Scheduling/CronTests.cs]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/elsa-workflows/elsa-core/pull/7036#discussion_r2498317466)
